### PR TITLE
M45 Trailblazer

### PR DESCRIPTION
**Pull request by Scarlet-Oduko**
## About The Pull Request 

https://user-images.githubusercontent.com/102184946/234426638-b1cc53ea-7cc8-4c55-92de-efac1122b7e7.mp4

Adds a new grenade, M45 Trailblazer, acts like a sticky grenade, sticking to xenos and humans alike, upon sticking as the mob moves they will leave behind a trail of fire until the grenade detonates, into a small area of fire.

## Why It's Good For The Game

Allows for creative use of fire and its ability to light up and clear mazes, denies xenos pheros and leaves a trail to track. Adds further interactions with fire based modules such as surt. interactions between xeno and marines, as the trail of fire can also be used on marines if shuffled into it. good source of good hits on both sides.

overall adds and expands strategies that marines can deploy and experiment with.
## Changelog
:cl:
add: M45 Trailblazer grenade
add: M45 Trailblazer grenade to marine vendor
add: M45 Trailblazer to req, purchasable in boxes
imageadd: adds M45 Trailblazer grenade icon and grenade box overlay
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 1]
```diff
@@ -111,6 +111,7 @@
 			/obj/item/explosive/grenade = 600,
 			/obj/item/explosive/grenade/m15 = 30,
 			/obj/item/explosive/grenade/sticky = 125,
+			/obj/item/explosive/grenade/sticky/trailblazer = 100,
```
Comment (novaepee) [inline code/game/objects/machinery/vending/marine_vending.dm:114]: This is my opinion, having a sticky incendiary shouldn't be more than incendiary. We will see whether xenomorphs cope and seeth hard.

**Comment by blackdav123** [comment]
fire has never been friendlier

excited to see what happens to a squad of marines as one of these is chucked at a charging crusher

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 1 above)
Comment (Scarlet-Oduko) [reply code/game/objects/machinery/vending/marine_vending.dm:114]: this was heavily based on stickies, so I reduced it a bit but yeah, if this gets TMed or merged. if 100 is too many I'll bump it down to incendiary level or maybe lower if its really bad.

**Comment by Al-1ce** [comment]
I can imagine this backfiring if the xeno decides to run into the marines, but I still like the idea.

**Comment by Autisem** [comment]
maby do add you can resist out of the grenade, like imagine you die after running away because the grenade is stuck to you for 5-10 seconds

**Comment by Scarlet-Oduko** [comment]
> maby do add you can resist out of the grenade, like imagine you die after running away because the grenade is stuck to you for 5-10 seconds

this concern is already baked into the grenade, besides the initial and final bits of fire. unless the xeno doubles back onto the trail it makes, the spawned fire tiles won't damage or stack fire onto the xeno. if it did, then any caste would be shredded. but if the trail is seen to last too long, I will lower det time.

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 1 above)
Comment (Scarlet-Oduko) [reply code/game/objects/machinery/vending/marine_vending.dm]: Reduced a bit more to 75 to be a good middle ground for testing.

File: code/modules/reqs/supplypacks.dm [hunk 2]
```diff
@@ -704,6 +704,12 @@ EXPLOSIVES
 	contains = list(/obj/item/storage/box/visual/grenade/M15)
 	cost = 500
 
+/datum/supply_packs/explosives/explosives_M45
+	name = "M45 Trailblazer grenade box crate"
+	notes = "contains 25 grenades"
```
Comment (XSlayer300) [inline code/modules/reqs/supplypacks.dm:709]:
```suggestion
	notes = "Contains 25 grenades"
```

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 3]
```diff
@@ -112,6 +112,34 @@
...
+	icon_state = "grenade_sticky_fire"
+	item_state = "grenade_sticky_fire"
+	det_time = 5 SECONDS
+	light_impact_range = 1
+
+/obj/item/explosive/grenade/sticky/trailblazer/prime()
+	flame_radius(0.5, get_turf(src))
+	playsound(loc, "incendiary_explosion", 35)
+	qdel(src)
+	return ..()
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/marines.dm:127]:
```suggestion
	return ..()
```

File: code/game/objects/items/storage/boxes.dm [hunk 4]
```diff
@@ -1115,6 +1115,13 @@
 	spawn_type = /obj/item/explosive/grenade/incendiary
 	closed_overlay = "grenade_box_overlay_hidp"
 
+/obj/item/storage/box/visual/grenade/M45
```
Comment (TiviPlus) [inline code/game/objects/items/storage/boxes.dm:1118]: no numbers in typepaths pls

File: code/modules/reqs/supplypacks.dm [hunk 5]
```diff
@@ -704,6 +704,12 @@ EXPLOSIVES
 	contains = list(/obj/item/storage/box/visual/grenade/M15)
 	cost = 500
 
+/datum/supply_packs/explosives/explosives_M45
```
Comment (TiviPlus) [inline code/modules/reqs/supplypacks.dm:707]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**